stacks-signer/src/runloop.rs

```diff
@@ -459,29 +480,109 @@ impl<C: Coordinator> RunLoop<C> {
                 });
             return false;
         };
+
         if block_info.valid.is_none() {
             // We have not yet received validation from the stacks node. Cache the request and wait for validation
             debug!("We have yet to receive validation from the stacks node for a nonce request. Cache the nonce request and wait for block validation...");
-            block_info.nonce_request = Some(request.clone());
+            block_info.nonce_request = Some(nonce_request.clone());
             return false;
         }
-        Self::determine_vote(block_info, request, transactions);
+
+        Self::determine_vote(block_info, nonce_request);
         true
     }
 
+    /// Verify the transactions in a block are as expected
+    fn verify_transactions(&mut self, block: &NakamotoBlock) -> bool {
+        if let Ok(expected_transactions) = self.get_expected_transactions() {
+            // Ensure the block contains the transactions we expect
+            let missing_transactions = expected_transactions
+                .into_iter()
+                .filter_map(|tx| {
+                    if !block.txs.contains(&tx) {
```

Comment:
Blocks can contain north of 10,000 transactions. It might be worth building a hashset of the blocks' txids and checking that against the expected transaction's txid.

Reply:
That won't allow us to give a hint back to the miner, are we okay with that?

Reply:
Ah, I'm speaking specifically about the call to `block.txs.contains(&tx)`.  `block.txs` is a `Vec`, so `.contains()` runs in _O(n)_ time.  I'm just asking about speeding this up.